recipes/mimalloc/all/conanfile.py

```diff
@@ -0,0 +1,123 @@
+from conans import ConanFile, CMake, MSBuild, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+import os
+
+
+class MimallocConan(ConanFile):
+    name = "mimalloc"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/mimalloc"
+    description = "mimalloc is a compact general purpose allocator with excellent performance."
+    topics = ("libraries", "c", "c++", "malloc", "allocator")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fPIC": [True, False],
+               "shared": [True, False],
+               "override": [True, False]}
+    default_options = {"fPIC": True,
+                       "shared": False,
+                       "override": True}
+    generators = "cmake", "visual_studio"
+    exports_sources = "patches/**"
+    _cmake_i = None
+    _msbuild_i = None
+
+    @property
+    def _cmake_install_folder(self):
+        p1 = self.version.find(".")
+        if p1 != -1:
+            p2 = self.version.find(".", p1 + 1)
+            if p2 != -1:
+                return self.name + "-" + self.version[0:p2]
+            else:
+                return self.name + "-" + self.version
+        else:
+            return self.name + "-" + self.version
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _cmake(self):
+        if not self._cmake_i:
+            self._cmake_i = CMake(self)
+            self._cmake_i.definitions["MI_BUILD_TESTS"] = False
+            if self.options.shared:
+                self._cmake_i.definitions["MI_BUILD_SHARED"] = True
+                self._cmake_i.definitions["MI_BUILD_STATIC"] = False
+            else:
+                self._cmake_i.definitions["MI_BUILD_SHARED"] = False
+                self._cmake_i.definitions["MI_BUILD_STATIC"] = True
+            self._cmake_i.configure(source_folder=self._source_subfolder,
+                build_folder=self._build_subfolder)
```

Comment:
You need to create a `CMakeLists.txt` wrapper script that includes `conanbuildinfo.cmake`,
calls `conan_basic_setup()` and runs `add_subdirectory(source_subfolder)`.
Conan uses this to set and check compiler options.
Please check other recipes that have a `CMakeLists.txt` in their `all` subdirectory on how to do this.

Reply:
Done.